app/models/status.rb

```diff
@@ -96,12 +96,12 @@ class Status < ApplicationRecord
   scope :not_excluded_by_account, ->(account) { where.not(account_id: account.excluded_from_timeline_account_ids) }
   scope :not_domain_blocked_by_account, ->(account) { account.excluded_from_timeline_domains.blank? ? left_outer_joins(:account) : left_outer_joins(:account).where('accounts.domain IS NULL OR accounts.domain NOT IN (?)', account.excluded_from_timeline_domains) }
   scope :tagged_with_all, ->(tag_ids) {
-    Array(tag_ids).reduce(self) do |result, id|
+    Array(tag_ids).map(&:to_i).reduce(self) do |result, id|
```

Comment:
I don't know ruby but the previous here looks quite dangerous as far as I can tell. Are the tag_ids sanitized somewhere else or is there another reason why this is not an sql injection?

Reply:
The ids are not passed from user input

Reply:
Still prepared statements are not possible when combining queries?

Reply:
No. That's what `tagged_with` does (returning statuses that are tagged with any of the given tags). `tagged_with_all` needs to return  tagged with *all* the given tags, there might be better way to query that, but right now I don't know which, and `tag_id IN (:tag_ids)` isn't one of them. Though `tagged_with_none` can most probably be rewritten in a better way using `tag_id IN (:tag_ids)`.

Reply:
I have rewritten `tagged_with_none` in a way that seems slightly more efficient and does not involve creating names from any input.